code/game/objects/items/storage/boxes.dm

```diff
@@ -476,45 +405,85 @@
+	name = "generic box"
+	desc = "This box is able to hold a wide variety of supplies."
+	icon = 'icons/obj/items/storage/storage_boxes.dmi'
+	icon_state = "mag_box"
+	item_state = "mag_box"
+	w_class = WEIGHT_CLASS_HUGE
+	slowdown = 0.4 // Big unhandly box
 	max_w_class = 4
-	storage_slots = 30
-	max_storage_space = 60	//SMG and pistol sized (tiny and small) mags can fit all 30 slots, normal (LMG and AR) fit 20
+	storage_slots = 32 // 8 images x 4 items
+	max_storage_space = 64
 	use_to_pickup = TRUE
 	can_hold = list(
-		/obj/item/ammo_magazine/pistol,
-		/obj/item/ammo_magazine/smg,
-		/obj/item/ammo_magazine/rifle,
-		/obj/item/ammo_magazine/magnum,
-		/obj/item/ammo_magazine/revolver,
-		/obj/item/ammo_magazine/acp,
-		/obj/item/ammo_magazine/standard_lmg,
-		/obj/item/ammo_magazine/standard_smartmachinegun,
-		/obj/item/ammo_magazine/m412l1_hpr,
-		/obj/item/ammo_magazine/shotgun,
-		/obj/item/ammo_magazine/sniper,
+		/obj/item, //This box should normally be unobtainable so here we go
 	)
-
+	///Assoc list linking each item to it's corresponding icon_state for the overlays if the atom does not have a item_state_mini var.
+	var/list/assoc_overlay = list()
+	///Assoc list of how much weight every item type takes. Used to determine how many overlays to make.
+	var/list/contents_weight = list()
+	///Initial offset of the overlays.
+	var/overlay_pixel_x = 5
+	var/overlay_pixel_y = 11
+	///Amount of overlay spaces per axis.
+	var/amt_horizontal = 4
+	var/amt_vertical = 2
+	///Amount of pixels to shift each overlay around only on one axis.
+	var/shift_x = BOX_OVERLAY_SHIFT_X
+	var/shift_y = BOX_OVERLAY_SHIFT_Y
+	///Whether or not the box is deployed on the ground
 	var/deployed = FALSE
+	///Amount of different items in the box.
+	var/variety = 0
+	///Amount of weight a single overlay can cover.
+	var/overlay_w_class = 0
+	///Total max amount of overlay spaces
+	var/max_overlays = 0
+	///Overlay icon_state to display on the box when it is closed
+	var/closed_overlay
+
+/obj/item/storage/box/visual/Initialize(mapload, ...)
+	. = ..()
+	max_overlays = amt_horizontal * amt_vertical
+	overlay_w_class = FLOOR(max_storage_space / max_overlays, 1)
+	can_hold -= cant_hold //Have cant_hold actually have a use
+	update_icon_state() //Getting the closed_overlay onto it
 
-/obj/item/storage/box/ammo/update_icon_state()
-	if(!deployed)
-		icon_state = "[initial(icon_state)]"
-	else if(!(length(contents)))
-		icon_state = "[initial(icon_state)]_empty"
-	else if(deployed)
-		icon_state = "[initial(icon_state)]_deployed"
+/obj/item/storage/box/visual/examine(mob/user, distance, infix, suffix)
+	. = ..()
+	if (!deployed && !(loc == user)) //Closed and not in your possession
+		return
+	if(variety > max_overlays) //Too much shit inside, a literal clusterfuck of supplies
+		to_chat(user, "It's too cluttered with all of these supplies inside.")
+		return
+	if(variety <= 0) //empy
+		to_chat(user, "[src] is empty!")
+		return
+	to_chat(user, "Inside [src] you notice:")
```

Comment:
note examiners see the name already so you might want to make this say "it" instead